ktor-http/common/src/io/ktor/http/ContentTypes.kt

```diff
@@ -286,7 +286,7 @@ public fun ContentType.withCharset(charset: Charset): ContentType =
  * if [ContentType] is not ignored
  */
 public fun ContentType.withCharsetIfNeeded(charset: Charset): ContentType =
-    if (contentType.lowercase() == "application" && contentSubtype.lowercase() == "json") {
+    if (contentType.lowercase() != "text") {
```

Comment:
This is a workaround replaced with a more general workaround.

Reply:
No, it's not. Only text content type should have charset parameter.

Reply:
You right. Although, I don't like the method's name. `withCharsetIfApplicable` would be better.